### PR TITLE
attempted externalizaation of prompts

### LINKED ISSUE #123: prompt-utils
Writing the entire inquirer.js prompt syntax in every place where it is needed is tiresome and takes space. Thus i will be externalizing it.

### PR DESCRIPTION
**Pull request by eetupur**
Resolves: #123.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,12 +26,15 @@
     "yaml": "^1.10.0"
   },
   "devDependencies": {
+    "@types/inquirer": "^7.3.1",
```
Comment (veikkahonkanen) [inline package.json]:
```suggestion
```

What kind of denedency is "i"?

File: src/classes/prompt-utils.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  /**
+   * Prompts the user with a message, that accepts text based input
+   * 
+   * @param message the message to show to the user
+   * @returns the user input
+   */
+  static inputPrompt = async (message : string) : Promise<string | null> => {
+    try {
+      console.log(3);
+      const prompt = await inquirer.prompt({
```
Comment (veikkahonkanen) [inline src/classes/prompt-utils.ts]:
```suggestion
  static inputPrompt = async (instance: any, message : string) : Promise<string | null> => {
    try {
      console.log(3);
      const prompt = await instance.prompt({
```

File: src/classes/prompt-utils.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  /**
+   * Prompts the user with a message, that offers a yes/no choise for input
+   * 
+   * @param message the message to show to the user
+   * @returns users input (string containing either "yes" or "no");
+   */
+  static confirmPrompt = async (message : string): Promise<string | null> => {
+    try {
+      const prompt = await inquirer.prompt({
```
Comment (veikkahonkanen) [inline src/classes/prompt-utils.ts]:
```suggestion
  static confirmPrompt = async (instance: any, message : string): Promise<string | null> => {
    try {
      const prompt = await instance.prompt({
```

File: src/classes/prompt-utils.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  /**
+   * Prompts the user with a message, that offers given choises as inputs
+   * 
+   * @param message the message to show to the user
+   * @param choices array of choises to show to the user
+   * @returns the choise that the user chose
+   */
+  static listPrompt = async (message : string, choices : string[]): Promise<string | null> => {
+    try {
+      const prompt = await inquirer.prompt({
```
Comment (veikkahonkanen) [inline src/classes/prompt-utils.ts]:
```suggestion
  static listPrompt = async (instance: any, message : string, choices : string[]): Promise<string | null> => {
    try {
      const prompt = await instance.prompt({
```

File: src/commands/new-repo.ts [hunk 5]
```diff
@@ -33,34 +34,25 @@ async function action(args) {
...
-        const nameResult = await this.prompt({
-          type: "input",
-          name: "name",
-          message: "Give a name for the repository (leave empty to cancel): "
-        });
-  
-        if (nameResult.name) {
-          repoName = nameResult.name;
+        console.log(1);
+        const nameResult = await PromptUtils.inputPrompt("Give a name for the repository (leave empty to cancel): ");
```
Comment (veikkahonkanen) [inline src/commands/new-repo.ts]:
```suggestion
        const nameResult = await PromptUtils.inputPrompt(this, "Give a name for the repository (leave empty to cancel): ");
```

File: src/commands/new-repo.ts [hunk 6]
```diff
@@ -33,34 +34,25 @@ async function action(args) {
...
         }
       }
 
-      const publicityResult = await this.prompt({
-        type: "list",
-        name: "publicity",
-        choices: [ "private", "internal", "public" ],
-        message: "Set the publicity of the repository: "
-      });
+      const publicityResult = await PromptUtils.listPrompt("Set the publicity of the repository: ", [ "private", "internal", "public" ]);
```
Comment (veikkahonkanen) [inline src/commands/new-repo.ts]:
```suggestion
      const publicityResult = await PromptUtils.listPrompt(this, "Set the publicity of the repository: ", [ "private", "internal", "public" ]);
```

File: src/commands/new-repo.ts [hunk 7]
```diff
@@ -33,34 +34,25 @@ async function action(args) {
...
   
-      const descriptionResult = await this.prompt({
-        type: "input",
-        name: "description",
-        message: "Give a description for the repository: "
-      });
+      const descriptionResult = await PromptUtils.inputPrompt("Give a description for the repository: ");
```
Comment (veikkahonkanen) [inline src/commands/new-repo.ts]:
```suggestion
      const descriptionResult = await PromptUtils.inputPrompt(this, "Give a description for the repository: ");
```

**Review by veikkahonkanen — CHANGES_REQUESTED** [review]
Use Visitor

File: package-lock.json [hunk 8]
```diff
@@ -816,219 +683,223 @@
...
+    "@types/inquirer": {
+      "version": "7.3.1",
+      "resolved": "https://registry.npmjs.org/@types/inquirer/-/inquirer-7.3.1.tgz",
+      "integrity": "sha512-osD38QVIfcdgsPCT0V3lD7eH0OFurX71Jft18bZrsVQWVRt6TuxRzlr0GJLrxoHZR2V5ph7/qP8se/dcnI7o0g==",
+      "dev": true,
+      "requires": {
+        "@types/through": "*",
+        "rxjs": "^6.4.0"
+      }
+    },
```
Comment (veikkahonkanen) [inline package-lock.json]:
```suggestion
```

File: package.json [hunk 9]
```diff
@@ -31,7 +31,9 @@
     "@typescript-eslint/parser": "^4.13.0",
     "coveralls": "^3.1.0",
     "eslint": "^7.17.0",
+    "i": "^0.3.6",
```
Comment (veikkahonkanen) [inline package.json]:
```suggestion
```

File: src/classes/prompt-utils.ts [hunk 10]
```diff
@@ -0,0 +1,74 @@
+import inquirer from "inquirer";
```
Comment (veikkahonkanen) [inline src/classes/prompt-utils.ts]:
```suggestion
```

File: src/classes/prompt-utils.ts [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+ * Provides functions for prompting user, helps shorten syntax
+ */
+export class PromptUtils {
+  /**
+   * Prompts the user with a message, that accepts text based input
+   * 
+   * @param message the message to show to the user
+   * @returns the user input
+   */
+  static inputPrompt = async (instance: any, message : string) : Promise<string | null> => {
```
Comment (veikkahonkanen) [inline src/classes/prompt-utils.ts]:
```suggestion
  static inputPrompt = async (instance: any, message: string): Promise<string | null> => {
```

File: src/classes/prompt-utils.ts [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+    }
+  }
+
+  /**
+   * Prompts the user with a message, that offers a yes/no choise for input
+   * 
+   * @param message the message to show to the user
+   * @returns users input (string containing either "yes" or "no");
+   */
+  static confirmPrompt = async (instance: any, message : string): Promise<string | null> => {
```
Comment (veikkahonkanen) [inline src/classes/prompt-utils.ts]:
```suggestion
  static confirmPrompt = async (instance: any, message: string): Promise<string | null> => {
```

File: src/classes/prompt-utils.ts [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  /**
+   * Prompts the user with a message, that offers given choises as inputs
+   * 
+   * @param message the message to show to the user
+   * @param choices array of choises to show to the user
+   * @returns the choise that the user chose
+   */
+  static listPrompt = async (instance: any, message : string, choices : string[]): Promise<string | null> => {
```
Comment (veikkahonkanen) [inline src/classes/prompt-utils.ts]:
```suggestion
  static listPrompt = async (instance: any, message: string, choices: string[]): Promise<string | null> => {
```

File: src/classes/prompt-utils.ts [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+    try {
+      const prompt = await instance.prompt({
+        type: "list",
+        name: "output",
+        message: message,
+        choices: choices
+      });
+
+      return prompt.output;
+
```
Comment (anttileppa) [inline src/classes/prompt-utils.ts]:
```suggestion
```

File: src/classes/prompt-utils.ts [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+  static confirmPrompt = async (instance: any, message: string): Promise<string | null> => {
+    try {
+      const prompt = await instance.prompt({
+        type: "confirm",
+        name: "output",
+        message: message
+      });
+
+      return prompt.output;
+
```
Comment (anttileppa) [inline src/classes/prompt-utils.ts]:
```suggestion
```

File: src/classes/prompt-utils.ts [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+  static inputPrompt = async (instance: any, message: string): Promise<string | null> => {
+    try {
+      const prompt = await instance.prompt({
+        type: "input",
+        name: "output",
+        message: message
+      });
+
+      return prompt.output;
+
```
Comment (anttileppa) [inline src/classes/prompt-utils.ts]:
```suggestion
```

File: package.json [hunk 17]
```diff
@@ -32,6 +32,7 @@
     "coveralls": "^3.1.0",
     "eslint": "^7.17.0",
     "jest": "^26.6.3",
+    "npm": "^7.6.2",
```
Comment (anttileppa) [inline package.json]: Why do you need npm as a dependecy?

**Review by anttileppa — CHANGES_REQUESTED** [review]

**Pull request merged**